### PR TITLE
[PIR][oneDNN] Add shuffle_channel_detect_pass

### PR DESCRIPTION
**Pull request by LLee233**
### PR Category

Others

### PR Types

New features

### Description

Based on new pass mechanism, here we add pass "shuffle_channel_detect_pass" for PIR.

The new pass is same as "shuffle_channel_onednn_detect_pass" in "/paddle/fluid/framework/ir/onednn/shuffle_channel_onednn_detect_pass.cc".

Note: add shuffle_channel/shuffle_channel_grad in ops.yaml/ops_backward.yaml to adapt PIR op.

### CODE REVIEW

**Review by yuanlehome — APPROVED** [review]

**Comment by LLee233** [comment]
@onecatcn Hi, would you mind helping find approvers as below? Since this PR includes OP API changes:
image

Thank you!

**Comment by onecatcn** [comment]
2024-05-10 13:32:59 **************************************************************
2024-05-10 13:32:59 Please find RD for approval first, and then find TPM for approval.
2024-05-10 13:32:59 0. You must have one RD (XiaoguangHu01, jeff41404, lanxianghit or qingqing01) approval for API change.
2024-05-10 13:32:59  and one TPM approval for API change: 
2024-05-10 13:32:59  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, Ligoml/LiMengLiu for general APIs.
2024-05-10 13:32:59 1. You must have one RD (XiaoguangHu01, jeff41404, lanxianghit or qingqing01) approval for API change.
2024-05-10 13:32:59  and one TPM approval for API change: 
2024-05-10 13:32:59  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, Ligoml/LiMengLiu for general APIs.
2024-05-10 13:32:59 2. You must have  one TPM approval for API documents change: 
2024-05-10 13:32:59  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, Ligoml/LiMengLiu for general API docs.
2024-05-10 13:32:59 
2024-05-10 13:32:59 There are 3 approved errors.
2024-05-10 13:32:59 **************************************************************

**Review by jzhang533 — APPROVED** [review]
LGTM for api changes

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**
